ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather.hpp

```diff
@@ -20,6 +20,17 @@ struct ExecuteAllGather {
         const std::optional<size_t> num_workers = std::nullopt,
         const std::optional<size_t> num_buffers_per_channel = std::nullopt,
         const ttnn::ccl::Topology topology = ttnn::ccl::Topology::Ring);
+
+    static ttnn::Tensor invoke(
+        const ttnn::Tensor& input_tensor,
+        const uint32_t dim,
+        const uint32_t cluster_axis,
+        const MeshDevice& mesh_device,
+        const uint32_t num_links = 1,
+        const std::optional<ttnn::MemoryConfig>& memory_config = std::nullopt,
+        const std::optional<size_t> num_workers = std::nullopt,
+        const std::optional<size_t> num_buffers_per_channel = std::nullopt,
+        const ttnn::ccl::Topology topology = ttnn::ccl::Topology::Linear);
```

Comment:
Sorry I wish I brought this up earlier but I think we should just get rid of this default. Especially since it differs from the default topology of the other overload, this will be very confusing to users and will probably lead to developers shooting themselves in the foot. For now we should atleast keep the default consistent.